server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -16,37 +16,48 @@ class Api::V21::DeploymentsController < Api::V2::DeploymentsController
    before_filter :find_deployment, :only => [:destroy, :show, :update, :deploy]
 
     def index
-      puts "XXX v21 index called"
-      super
+      @deployments = Deployment.all
+      render :json => @deployments, :each_serializer => Fusor::DeploymentSerializer
+    end
+
+    def show
+      render :json => @deployment, :serializer => Fusor::DeploymentSerializer
+    end
+
+    def create
+      @deployment = Deployment.new(params[:hostgroup])
+      @deployment.save
+      if @deployment.save
+        render :json => @deployment, :serializer => Fusor::DeploymentSerializer
+      else
+        render json: {errors: @deployment.errors}, status: 422
+      end
+    end
+
+    def update
+      if @deployment.update_attributes(params[:deployment])
+        render :json => @deployment, :serializer => Fusor::DeploymentSerializer
+      else
+        render json: {errors: @deployment.errors}, status: 422
+      end
+    end
+
+    def destroy
+      @deployment.destroy
+      respond_for_show :resource => @deployment
+    end
+
+    def deploy
+      task = async_task(::Actions::Fusor::Deploy, @deployment)
+      respond_for_async :resource => task
+    end
+
+    private
+
+    def find_deployment
+      not_found and return false if params[:id].blank?
+      @deployment = Deployment.find(params[:id])
```

Comment:
Is there a way to have this controller extend the V2 so we don't have to reimplement all of the methods? That's what I was trying to do but I couldn't get it to work. Hence the ```super``` call in the old ```index``` method. I don't mind having 2 versions just a bit scared of maintaining both down the road.